src/command_modules/azure-cli-acs/azure/cli/command_modules/acs/custom.py

```diff
@@ -1418,6 +1418,122 @@ def aks_upgrade(cmd, client, resource_group_name, name, kubernetes_version, no_w
+        import urllib.request
+        urllib.request.urlretrieve(setup_url, setup_file)
+        try:
+            subprocess.call(
+                setup_args, universal_newlines=True, stdin=None, stdout=None, stderr=None, shell=False)
+            os.remove(setup_file)
+        except subprocess.CalledProcessError as err:
+            raise CLIError('Could not install {}: {}'.format(vsce_tool, err))
+        except PermissionError:
+            raise CLIError('Installing {} tooling needs permissions.'.format(vsce_tool))
+        try:
+            _ensure_dev_connected_installed(vsce_cli)
+        except OSError:
+            raise CLIError("{} not installed properly. Use 'az aks use-dev-connect' commands for \
+            connected development.".format(vsce_tool))
+
+    create_dev = False
+    try:
+        from subprocess import PIPE
+        retCode = subprocess.call(
+            [vsce_cli, 'env', 'select', '-n', cluster_name, '-g', resource_group_name],
+            stderr=PIPE)
+        if retCode == 1:
+            create_dev = True
+    except subprocess.CalledProcessError as err:
+        create_dev = True
+
+    if create_dev:
+        try:
+            subprocess.call(
+                [vsce_cli, 'env', 'create', '--aks-name', cluster_name, '--aks-resource-group', 
+                resource_group_name], 
+                universal_newlines=True)
+        except subprocess.CalledProcessError as err:
+            raise CLIError('{} creation failure: {}.'.format(vsce_tool, err))
+
+
+def aks_remove_devconnect(cmd, client, cluster_name, resource_group_name, prompt=False):  # pylint: disable=line-too-long
+    """
+    Remove Azure Dev Connect from a managed Kubernetes cluster.
+
+    :param cluster_name: Name of the target AKS cluster.
+    :type cluster_name: String
+    :param resource_group_name: Name of the target AKS cluster's resource group.
+    :type resource_group_name: String
+    :param prompt: Do not prompt for confirmation.
+    :type prompt: bool
+    """
+
+    vsce_tool = 'Visual Studio Connected Development Services'
+    vsce_cli = 'vsce'
+
+    try:
+        _ensure_dev_connected_installed(vsce_cli)
+    except OSError:
+        raise CLIError("{} not detected, please verify if it is installed. Use 'az aks use-dev-connect' \
+        commands for connected development.".format(vsce_tool))
+
+    remove_command_arguments = [vsce_cli, 'env', 'rm', '--name', 
+                                cluster_name, '--resource-group', 
+                                resource_group_name]
+
+    if prompt:
+        remove_command_arguments.append('-f')
+    try:
+        subprocess.call(
+            remove_command_arguments, universal_newlines=True)
+    except subprocess.CalledProcessError as err:
+        raise CLIError('{} deletion failure: {}.'.format(vsce_tool, err))
+
+
+def _ensure_dev_connected_installed(vsce_cli):
+    from subprocess import PIPE, Popen
+    Popen([vsce_cli], stdout=PIPE, stderr=PIPE)
```

Comment:
This function doesn't actually ensure that it is installed.
Suggest renaming this to `_is_dev_connected_installed()` then return True or False only.

Reply:
Shouldn't it be `[_ensure|_is]_dev_connect_installed`? Not sure why `connected` is in there.

Reply:
Yes. I don't remember why I named it 'connected'. Will update it. Thanks